File: Frontend/fpams-web/src/types/facultyProfile.ts
export interface FacultyProfile {
    id: string;
    userId: string;
    employeeCode: string;
    facultyName: string;
    email: string;
    departmentId: string;
    departmentName: string;
    designationId: string;
    designationName: string;
    dateOfJoining: string;
    highestQualification: string;
    specialization: string;
    teachingExperienceYears: number;
    industryExperienceYears: number;
    isActive: boolean;
}

export interface CreateFacultyProfileRequest {
    userId: string;
    departmentId: string;
    designationId: string;
    dateOfJoining: string;
    highestQualification: string;
    specialization: string;
    teachingExperienceYears: number;
    industryExperienceYears: number;
}

export interface UpdateFacultyProfileRequest extends CreateFacultyProfileRequest {
    id: string;
    isActive: boolean;
}

File: Backend/FPAMS/FPAMS.Domain/Entities/FacultyProfile.cs
using FPAMS.Domain.Common;

namespace FPAMS.Domain.Entities;

public class FacultyProfile : BaseEntity
{
    public Guid UserId { get; set; }

    public User User { get; set; } = null!;

    public Guid DepartmentId { get; set; }

    public Department Department { get; set; } = null!;

    public Guid DesignationId { get; set; }

    public Designation Designation { get; set; } = null!;

    public DateTime DateOfJoining { get; set; }

    public string HighestQualification { get; set; } = string.Empty;

    public string Specialization { get; set; } = string.Empty;

    public int TeachingExperienceYears { get; set; }

    public int IndustryExperienceYears { get; set; }

    public bool IsActive { get; set; } = true;
}
